fixtures/futures/src/lib.rs

```diff
@@ -276,4 +279,54 @@ pub async fn broken_sleep(ms: u16, fail_after: u16) {
     .await;
 }
 
+#[derive(uniffi::Object)]
+/// Test scheduling with a foreign executor
+pub struct SchedulingTester {
+    background_executor: uniffi::ForeignExecutor,
+    counter: AtomicI32,
+}
+
+#[uniffi::export]
+impl SchedulingTester {
+    #[uniffi::constructor]
+    pub fn new(background_executor: uniffi::ForeignExecutor) -> Arc<Self> {
+        Arc::new(Self {
+            background_executor,
+            counter: AtomicI32::new(0),
+        })
+    }
+
+    /// Simulate a long-running calculation that's scheduled on a background thread via the foreign
+    /// executor.
+    ///
+    /// This tests the run() method which schedules a task to run "fire-and-forget" style and
+    /// returns a Future for the result
+    pub async fn run_calculation(self: Arc<Self>) -> i32 {
+        Arc::clone(&self)
+            .background_executor
+            .run(0, move || self.do_calculation())
+            .await
+    }
+
+    fn do_calculation(&self) -> i32 {
+        // Pretend this is code that would take a long time to execute and should be scheduled on a
+        // background thread
+        42
+    }
+
+    /// Schedule incrementing a counter value using the foreign executor
+    ///
+    /// This tests the schedule() method which schedules a task to run "fire-and-forget" style,
+    /// where we don't get the result back
+    pub fn schedule_increment(self: Arc<Self>) {
+        Arc::clone(&self).background_executor.schedule(0, move || {
```

Comment:
The `Arc::clone()` call is not really needed, but I couldn't find a way around it.  If you change it to `self.background_executor.schedule()` then Rust complains because the closure moves `self` while it's still borrowed for the `schedule()` call.

In practice I don't think there's any threading issues, because the closure won't execute until later on, but I can't figure out a way to tell Rust that.